aml/org.testeditor.aml.dsl/src/org/testeditor/aml/dsl/validation/AmlValidator.xtend

```diff
@@ -180,6 +187,43 @@ class AmlValidator extends AbstractAmlValidator {
 			]
 		]
 	}
+	
+	static val ID_REGEX='\\^?[a-zA-Z$_][a-zA-Z$_0-9]*' 
+	static val VALID_TEMPLATE_WORDS = '''^[ \t]*(«ID_REGEX»[ \t]*)+$'''
+	static val VALID_LAST_TEMPLATE_WORDS = '''^[ \t]*(«ID_REGEX»[ \t]*)*([.?][ \t]*)?$'''
+	
+	/**
+	 * This validation is tightly coupled with the parser rules for tcl-template usages (see Tcl.xtext:TestStep and Tcl.xtext:TestStepAssignment)
+	 */
+	@Check
+	def void checkTemplateHoldsValidCharacters(Template template) {
+		template.contents => [
+			// all text elements must not be empty (checked here to allow index-based error marking)
+			val templateTextsThatAreEmpty = indexed.filterValue(TemplateText).filter[value.value.trim.empty]
```

Comment:
The slection in the list with two filter steps was not as easy to read it the first time